assert/require binary output does not escape %s

I am using `{require/assert}.Equal` on random binary bytes and every now and then they contain sequences like %v, %s and those cause output like `%..(MISSING)`.

I bet they can be escaped. Here's some output:

```
            Diff:
            --- Expected
            +++ Actual
            @@ -1,66 +1,66 @@
             ([]uint8) (len=1024 cap=1536) {
            - 00000000  e2 54 43 61 a9 15 d5 d1  8c b8 68 7f 6e 2c d7 32  |.TCa......h.n,.2|
            - 00000010  6d 92 3c 21 29 46 74 e9  b6 88 6b ba 26 c0 40 0c  |m.<!)Ft...k.&.@.|
            - 00000020  67 99 f1 09 ef 35 59 6c  3f d9 37 cf b2 4d b8 57  |g....5Yl?.7..M.W|
            - 00000030  74 ce 94 c4 0d b5 e7 79  67 a5 43 ae e1 4d 12 43  |t......yg.C..M.C|
            - 00000040  5e c3 9c f5 38 36 87 43  c0 9a b2 82 0a 19 19 1f  |^...86.C........|
            - 00000050  de 75 cf 0e a8 b1 1e d1  ed 3a 02 c2 fa 16 dd 84  |.u.......:......|
            - 00000060  5b c4 0d a4 32 cf 81 9c  ef 97 65 33 54 19 f8 5d  |[...2.....e3T..]|
            - 00000070  bf 52 69 16 14 6f 57 63  34 16 0f 14 cd e6 e0 63  |.Ri..oWc4......c|
            - 00000080  86 83 ed e4 5c e2 04 33  cc 8d 4e 91 24 f2 c5 0d  |....\..3..N.$...|
            - 00000090  4d 83 de 59 6b 7c c1 8a  e2 8d 8e 0d a7 51 a9 34  |M..Yk|.......Q.4|
            - 000000a0  cb 05 16 e1 26 cd ba 86  3c ad 6c c8 43 a6 94 e3  |....&...<.l.C...|
            - 000000b0  cb fe f9 37 d6 30 d6 87  5f b5 e3 55 17 fa d5 75  |...7.0.._..U...u|
            - 000000c0  36 b5 2c 5f ef 1d f6 5e  68 f0 4c b3 94 2c 66 b1  |6.,_...^h.L..,f.|
            - 000000d0  76 db e7 51 d7 23 50 1e  d3 a5 a3 33 24 bc 64 36  |v..Q.#P....3$.d6|
            - 000000e0  82 64 02 3c 4f 84 de 5b  65 48 38 eb c4 56 14 81  |.d.<O..[eH8..V..|
            - 000000f0  e8 71 2b 61 7a 01 78 a8  9d e5 ab 3f 2d 5c 0f a2  |.q+az.x....?-\..|
            - 00000100  0a 2b 37 d4 d6 48 88 42  46 0a f3 30 7a 30 f7 f2  |.+7..H.BF..0z0..|
            - 00000110  ce 53 20 cd 1e 84 aa f2  50 ad d2 32 b4 a5 45 a4  |.S .....P..2..E.|
            - 00000120  82 20 8f b7 1f 30 52 5d  b8 13 e4 d4 ec 50 c6 0e  |. ...0R].....P..|
            - 00000130  d8 d8 a9 b6 ef ba 60 39  da cc ac dc 2d 5c 0c c9  |......`9....-\..|
            - 00000140  2e 81 1a e1 9e 4f 76 13  30 8f 0f ff 79 ce ee 24  |.....Ov.0...y..$|
            - 00000150  ec 17 a2 81 5a a7 98 ff  49 46 b4 4d 36 0f 5e 88  |....Z...IF.M6.^.|
            - 00000160  03 34 fb 8b 4d 6f 6b 10  4e 92 0d 6d 87 c1 29 a1  |.4..Mok.N..m..).|
            - 00000170  38 17 4d a7 2e be 43 aa  e3 59 d9 ae 8f 47 d0 65  |8.M...C..Y...G.e|
            - 00000180  0b f6 1a 6e 52 93 d5 cd  10 15 a7 c5 05 c2 55 c3  |...nR.........U.|
            - 00000190  ea 30 53 87 4a 6d 17 4c  ee 5e bb 85 00 7c db f9  |.0S.Jm.L.^...|..|
            - 000001a0  07 ce 0c d2 87 3c a9 76  d9 9b 22 ce 1c e6 7e 10  |.....<.v.."...~.|
            - 000001b0  0b ab 6f 64 15 14 27 62  ca 85 64 fe 1c bc ea 59  |..od..'b..d....Y|
            - 000001c0  88 9c 86 76 cd e3 f6 39  b2 2c 09 ee f1 0e 96 09  |...v...9.,......|
            - 000001d0  0b ef 50 e3 9c 82 58 07  42 a8 79 c3 46 d8 c6 5a  |..P...X.B.y.F..Z|
            - 000001e0  99 fd cc 66 b5 eb 1e ef  48 11 3e cc d0 80 a6 7c  |...f....H.>....||
            - 000001f0  42 d1 f9 d9 62 63 8e 55  bb 65 35 22 3e f7 c3 c0  |B...bc.U.e5">...|
            - 00000200  aa 71 5f aa f3 ae 20 b7  c4 b6 17 2e 1d ce 25 9a  |.q_... .......%!|(MISSING)
            - 00000210  b7 bf dc c1 04 48 24 03  8c 67 1e 05 25 d4 3e a5  |.....H$..g..%!>(MISSING).|
            - 00000220  35 93 5f 12 b9 f0 0a 57  78 e5 d9 ca 86 f8 f1 31  |5._....Wx......1|
            - 00000230  86 17 43 83 5b d3 7a 72  f0 73 f2 ce e3 ef 84 ff  |..C.[.zr.s......|
            - 00000240  d8 e0 f3 c4 6c da fc f7  fe 0b 36 5a 80 2a b0 26  |....l.....6Z.*.&|
            - 00000250  a4 57 15 b4 c4 d0 86 d8  02 8b c3 56 25 2e 8d 4e  |.W.........V%!.(MISSING)N|
            - 00000260  a0 91 fb fe 87 0e 28 3c  2b b0 fb e8 04 8f f2 73  |......(<+......s|
            - 00000270  d0 96 74 d5 0e c7 61 7c  f8 e5 7b 60 16 32 21 07  |..t...a|..{`.2!.|
            - 00000280  1b 79 db 2f 97 ab d6 65  7a f1 b7 00 ef 44 f9 3e  |.y./...ez....D.>|
            - 00000290  70 de 7d 62 6d 29 4d 45  9e 71 44 4e 0e 70 f7 91  |p.}bm)ME.qDN.p..|
            - 000002a0  06 9a ff 46 51 ea 39 ce  b7 07 76 ba 82 98 35 27  |...FQ.9...v...5'|
            - 000002b0  8d 56 d7 0f 9f e8 eb ff  ff 04 5c bd df fd 19 f0  |.V........\.....|
            - 000002c0  5a 2e 83 e5 fc 3e 16 f5  93 99 d0 7c b0 84 66 64  |Z....>.....|..fd|
            - 000002d0  eb c4 b5 cd e6 43 fd 5a  d2 97 e7 f6 7f 47 98 1c  |.....C.Z.....G..|
            - 000002e0  13 3e ea 14 77 88 f3 7e  cb e7 9e 2b 92 be e7 38  |.>..w..~...+...8|
            - 000002f0  4b 5e f6 82 0b 18 01 bb  76 db f6 33 91 71 1e 01  |K^......v..3.q..|
            - 00000300  38 e7 82 77 26 e5 61 f9  e3 bf 81 e0 0a a7 b4 a4  |8..w&.a.........|
            - 00000310  e8 dd 6f e1 7e 2f e9 5c  8b 61 1a fa d6 ed 71 45  |..o.~/.\.a....qE|
            - 00000320  e4 f9 a6 ba fa 9c d9 93  42 e8 c8 93 44 9b a5 a2  |........B...D...|
            - 00000330  69 c5 84 19 50 95 7b f6  25 02 87 7d dd 04 cb 2c  |i...P.{.%!.(MISSING)}...,|
            - 00000340  06 5a 25 4e 8b e2 8d 86  f0 6d f0 25 6d 43 bb a6  |.Z%!N(MISSING).....m.%!m(MISSING)C..|
            - 00000350  4b e0 7e 65 9e 03 c4 da  18 69 75 60 52 33 c0 7d  |K.~e.....iu`R3.}|
            - 00000360  10 cf bf 82 9e 1b ef 04  2c 56 60 c7 fc 09 f5 44  |........,V`....D|
            - 00000370  91 3a 95 3b 70 84 ca d7  a3 d2 3b f7 6e db 3b 66  |.:.;p.....;.n.;f|
            - 00000380  8c d7 ee d5 08 7f 10 ee  82 3b cb d2 de 85 b9 df  |.........;......|
            - 00000390  a8 44 98 90 50 d1 bd ff  17 d4 8c 19 92 a8 18 b3  |.D..P...........|
            - 000003a0  c4 ab 35 20 e5 a1 9c 78  b4 b3 58 c6 ba 7c d9 ac  |..5 ...x..X..|..|
            - 000003b0  55 29 62 f1 3f 4e d1 68  f6 0e 72 22 fa 38 ff c3  |U)b.?N.h..r".8..|
            - 000003c0  cf 64 2b cc 7d 98 3d 53  bf 57 c8 85 94 7a c5 76  |.d+.}.=S.W...z.v|
            - 000003d0  b8 12 df 3f 33 a9 81 08  13 23 13 c9 ba a6 e0 7a  |...?3....#.....z|
            - 000003e0  bd b8 fd 6f f9 79 1d 41  e4 9b 54 44 58 0a 0b cc  |...o.y.A..TDX...|
            - 000003f0  3b 1c c1 1f e9 a7 30 f6  c9 ec 6b 22 51 65 47 d7  |;.....0...k"QeG.|
            + 00000000  32 53 d4 54 2c d4 86 75  ab c1 0a d2 cc 23 aa bf  |2S.T,..u.....#..|
            + 00000010  5b 8d f0 f3 b9 f6 13 20  b9 84 60 11 ee 0d 3b d9  |[...... ..`...;.|
            + 00000020  a0 fb 72 11 42 33 c7 b6  2d 9a 04 47 49 1b d4 8b  |..r.B3..-..GI...|
            + 00000030  8e db 6e 67 78 85 4b 32  a5 ef 43 53 1d f3 98 0c  |..ngx.K2..CS....|
            + 00000040  ee 23 e8 e2 c0 50 8f 74  a8 92 b8 38 49 c0 15 29  |.#...P.t...8I..)|
            + 00000050  09 a8 6e cf 3c 06 e8 de  79 ab 70 0c db 0a b6 a4  |..n.<...y.p.....|
            + 00000060  fe be 23 0f de cb 52 6b  d1 2a 11 41 5c 38 70 b3  |..#...Rk.*.A\8p.|
            + 00000070  6a b0 63 9a c4 16 fc 04  e4 a9 fa 66 b7 5d a4 11  |j.c........f.]..|
            + 00000080  ec 44 fc 2c 4a 35 07 65  c2 7f a0 a2 b2 82 6e 62  |.D.,J5.e......nb|
            + 00000090  86 25 4a be 3e c4 16 3b  95 7c b4 cf 33 c0 d4 42  |.%!J(MISSING).>..;.|..3..B|
            + 000000a0  cc f7 7a 9e a3 32 b3 01  85 5c e0 1f dd 55 9e a0  |..z..2...\...U..|
            + 000000b0  32 45 49 a9 4b 7d ed 59  bb 6a 95 74 53 05 e9 87  |2EI.K}.Y.j.tS...|
            + 000000c0  bc 03 6c 20 7a af 3f c5  44 f8 ed 0f 71 d3 34 b7  |..l z.?.D...q.4.|
            + 000000d0  10 4b 53 cc bc f6 f6 67  44 81 77 e9 5f 84 3f 93  |.KS....gD.w._.?.|
            + 000000e0  da 02 78 1f b4 46 10 3f  56 2d 72 2b ef 47 b5 28  |..x..F.?V-r+.G.(|
            + 000000f0  ef 24 2b 44 81 24 04 e2  2d 74 34 3a 7f f1 e9 99  |.$+D.$..-t4:....|
            + 00000100  c8 79 ba 4d 7d 2c fa 70  91 93 fe aa 20 d6 14 70  |.y.M},.p.... ..p|
            + 00000110  ca 90 58 89 64 70 6e 07  56 78 51 b4 a5 80 5d c3  |..X.dpn.VxQ...].|
            + 00000120  46 19 7a 46 f5 50 f0 7f  f1 77 b6 39 14 32 3b 66  |F.zF.P...w.9.2;f|
            + 00000130  c6 16 20 d4 e3 1a 0b a7  3f ed e5 6e 79 40 97 a8  |.. .....?..ny@..|
            + 00000140  ae 90 00 95 01 d2 32 f6  6e f9 b1 6d c6 90 87 2a  |......2.n..m...*|
            + 00000150  f6 ff ff d4 08 9f 65 95  94 2e 8c fc ea 48 8b bf  |......e......H..|
            + 00000160  48 69 50 cb dd 1a be 40  e7 ab 97 f8 bf 5b ff 02  |HiP....@.....[..|
            + 00000170  43 38 3a 0d 32 d7 3b 84  cc 96 8f 73 b2 29 56 87  |C8:.2.;....s.)V.|
            + 00000180  45 80 d2 1c b3 29 26 b0  cd ab 9c cc 68 dc f8 41  |E....)&.....h..A|
            + 00000190  a6 71 57 10 64 8b 99 ff  00 91 0d 89 45 0d f3 44  |.qW.d.......E..D|
            + 000001a0  72 9a 30 51 80 57 b8 d9  a0 6c 74 28 db 0d fa 2b  |r.0Q.W...lt(...+|
            + 000001b0  b8 ae 22 8c 7e d2 55 b0  1d 94 b6 ef 09 86 af 5e  |..".~.U........^|
            + 000001c0  f0 82 c2 80 e8 34 6c 9b  ea 13 88 d4 db 1b cd fa  |.....4l.........|
            + 000001d0  42 a9 2c 59 f2 01 94 7b  3b 4c d4 cb 19 c4 b1 1e  |B.,Y...{;L......|
            + 000001e0  08 9f a2 f3 b0 4a d1 17  da 8e a4 60 3b 1b d1 e8  |.....J.....`;...|
            + 000001f0  d4 b2 2e ea 3c 6a 2c 81  e3 85 a1 21 63 43 78 0d  |....<j,....!cCx.|
            + 00000200  3a 51 df 29 02 ca d4 fc  22 3c 1b 2b df e3 6e 6c  |:Q.)...."<.+..nl|
            + 00000210  50 da 05 a0 a5 a6 36 a1  b5 1e 1c 12 30 fd 1f 87  |P.....6.....0...|
            + 00000220  cb 4a b5 0a ac bf 82 5d  f0 f4 77 6b 8c 06 dd d6  |.J.....]..wk....|
            + 00000230  32 75 8b 3c 54 36 e8 a9  65 a6 dc df 22 8f 40 78  |2u.<T6..e...".@x|
            + 00000240  98 fb ff 0e 84 ae 6a f2  77 4a a9 26 91 76 79 fd  |......j.wJ.&.vy.|
            + 00000250  c9 96 50 3d 37 bc 9c da  6c c6 fc 84 d7 03 76 97  |..P=7...l.....v.|
            + 00000260  4a 30 5e 4e 0c 05 77 e0  a0 7a cd 2a 72 d3 ce 2e  |J0^N..w..z.*r...|
            + 00000270  9b fe 58 e6 1f ea 19 5a  30 bf 0a 23 20 77 db 12  |..X....Z0..# w..|
            + 00000280  28 45 e6 4c be d3 37 57  7f bd bb 2b 30 5c 38 fd  |(E.L..7W...+0\8.|
            + 00000290  fd ee 04 5e 81 f3 99 93  9d 8f f3 30 e7 ee 41 8f  |...^.......0..A.|
            + 000002a0  a9 70 5a c3 70 ac 80 e3  f7 d1 93 0d 71 7a 33 cd  |.pZ.p.......qz3.|
            + 000002b0  93 92 0e 9b df 37 64 88  ae 2c bb b9 a7 c9 dc c9  |.....7d..,......|
            + 000002c0  5b 7d f4 85 8c 2c 34 06  d8 5d 55 f0 73 56 c8 d7  |[}...,4..]U.sV..|
            + 000002d0  e3 72 3b 4d c9 76 04 6c  44 b8 d1 8c ff ab aa f9  |.r;M.v.lD.......|
            + 000002e0  e6 13 54 16 43 64 67 3d  4b 4a 71 d0 34 6a a1 f8  |..T.Cdg=KJq.4j..|
            + 000002f0  45 42 c1 74 51 b0 7b 24  e2 10 63 a2 5d 38 8b 74  |EB.tQ.{$..c.]8.t|
            + 00000300  66 09 2f f4 9d c6 e0 84  d7 d0 b8 45 5d 76 a8 f0  |f./........E]v..|
            + 00000310  fd 31 2b a8 65 53 6f 7f  20 f7 ad 63 86 f0 84 73  |.1+.eSo. ..c...s|
            + 00000320  59 90 e0 39 e6 ea c3 8a  41 01 96 df 33 cb 4e 4d  |Y..9....A...3.NM|
            + 00000330  e7 73 c2 6f 09 93 4a 08  ae 36 dd 8b 72 3f 08 25  |.s.o..J..6..r?.%!|(MISSING)
            + 00000340  3e 0a 6e 61 9d 7d 4f 6c  ef 58 43 18 cf 57 27 b3  |>.na.}Ol.XC..W'.|
            + 00000350  e2 be 56 78 e4 d4 8c 58  8e c7 51 7c b7 04 10 53  |..Vx...X..Q|...S|
            + 00000360  86 b3 9f ff 76 3a 8a d3  e0 c7 3c 80 bd 7b 2b fe  |....v:....<..{+.|
            + 00000370  b7 a7 69 cf c3 2b 47 29  cc d1 8e d0 21 1c 2c 23  |..i..+G)....!.,#|
            + 00000380  4e 44 3c 7d f3 d8 18 00  87 63 04 2e 57 dc 34 83  |ND<}.....c..W.4.|
            + 00000390  04 f6 da 24 94 51 d7 8f  ce 6e 85 56 52 97 9f 3d  |...$.Q...n.VR..=|
            + 000003a0  6b 33 84 b6 14 90 4d 2a  6d 2c e4 b2 83 4c 36 03  |k3....M*m,...L6.|
            + 000003b0  e7 48 bc 63 a9 dd c3 28  85 ac ed a1 9c 5d 9e 6a  |.H.c...(.....].j|
            + 000003c0  ef 43 df 9c 9a bb 75 97  46 dc 0b ae 05 ed 47 8a  |.C....u.F.....G.|
            + 000003d0  b6 6f 04 80 dc ac 44 5e  0c 47 8f a6 3d a7 8e 53  |.o....D^.G..=..S|
            + 000003e0  e5 16 8e d8 7e ab b6 e3  7f 15 4c 35 a0 8c d4 f0  |....~.....L5....|
            + 000003f0  52 db 82 dc 40 df 48 0c  5a 20 26 01 ca f9 a8 99  |R...@.H.Z &.....|
```

PR submitted.  Please review.

The following simple test triggers this bug:

```go
func Test325(t *testing.T) {
	assert.Equal(t, "%%26", "%26")
}
```

Received Output:

```text
--- FAIL: Test325 (0.00s)
        Error Trace:    collections_test.go:91
        Error:          Not equal: 
                        expected: "%26"
                        received: "%!"(MISSING)
```

Expected Output:

```text
--- FAIL: Test325 (0.00s)
        Error Trace:    collections_test.go:91
        Error:          Not equal: 
                        expected: "%%26"
                        received: "%26"
```

--- a/assert/assertions.go
+++ b/assert/assertions.go
@@ -181,7 +181,7 @@ func indentMessageLines(message string, longestLabelLen int) string {
 		// no need to align first line because it starts at the correct location (after the label)
 		if i != 0 {
 			// append alignLen+1 spaces to align with "{{longestLabel}}:" before adding tab
-			outBuf.WriteString("\n\r\t" + strings.Repeat(" ", longestLabelLen +1) + "\t")
+			outBuf.WriteString("\n\r\t" + strings.Repeat(" ", longestLabelLen+1) + "\t")
 		}
 		outBuf.WriteString(scanner.Text())
 	}
@@ -223,13 +223,13 @@ func Fail(t TestingT, failureMessage string, msgAndArgs ...interface{}) bool {
 		content = append(content, labeledContent{"Messages", message})
 	}
 
-	t.Errorf("\r" + getWhitespaceString() + labeledOutput(content...))
+	t.Errorf("%s", "\r"+getWhitespaceString()+labeledOutput(content...))
 
 	return false
 }
 
 type labeledContent struct {
-	label string
+	label   string
 	content string
 }
 

--- a/assert/assertions_test.go
+++ b/assert/assertions_test.go
@@ -630,7 +630,7 @@ func TestNoError(t *testing.T) {
 	}()
 
 	if err == nil { // err is not nil here!
-		t.Errorf("Error should be nil due to empty interface", err)
+		t.Error("Error should be nil due to empty interface", err)
 	}
 
 	False(t, NoError(mockT, err), "NoError should fail with empty error interface")
@@ -664,7 +664,7 @@ func TestError(t *testing.T) {
 	}()
 
 	if err == nil { // err is not nil here!
-		t.Errorf("Error should be nil due to empty interface", err)
+		t.Error("Error should be nil due to empty interface", err)
 	}
 
 	True(t, Error(mockT, err), "Error should pass with empty error interface")

